packages/strapi-plugin-upload/services/Upload.js

```diff
@@ -33,27 +35,25 @@ module.exports = {
           ext: _.last(stream.name.split('.')),
           buffer: Buffer.concat(buffers),
           mime: stream.type,
-          size: `${(stream.size / 1000).toFixed(2)} KB`
+          size: (stream.size / 1000).toFixed(2)
         };
       })
     );
   },
 
-  upload: (files) => {
+  upload: async (files, config) => {
+    // get upload provider settings to configure the provider to use
+    const provider = _.cloneDeep(_.find(strapi.plugins.upload.config.providers, {provider: config.provider}));
```

Comment:
It seems not dynamic enough. In this function (https://github.com/strapi/strapi/pull/684/files#diff-163ee08ebcb2320b4f4805a7d02ed6daR74), when we update the settings for example (the AWS S3 bucket name) the changes are stored in the database. But the `strapi.plugins.upload.config` is not refreshed because the server hasn't restarted. So it means that to work as excepted I have to restart my server to use the latest configurations which is not the goal to store these settings in the database.

Reply:
`strapi.plugins.upload.config.providers` is the list of installed providers.